vsintegration/src/FSharp.Editor/Common/LanguageService.fs

```diff
@@ -250,28 +250,25 @@ and
     internal FSharpLanguageService(package : FSharpPackage) =
     inherit AbstractLanguageService<FSharpPackage, FSharpLanguageService>(package)
 
-    let checkerProvider = package.ComponentModel.DefaultExportProvider.GetExport<FSharpCheckerProvider>().Value
     let projectInfoManager = package.ComponentModel.DefaultExportProvider.GetExport<ProjectInfoManager>().Value
 
     let projectDisplayNameOf projectFileName = 
         if String.IsNullOrWhiteSpace projectFileName then projectFileName
         else Path.GetFileNameWithoutExtension projectFileName
 
-    let openedProjects = Queue<IProvideProjectSite>()
-
-    do
-      Events.SolutionEvents.OnAfterOpenProject.Add (fun (args: Events.OpenProjectEventArgs) -> 
-        match args.Hierarchy with
-        | :? IProvideProjectSite as siteProvider ->
-            openedProjects.Enqueue(siteProvider)            
-        | _ -> ())
-        
-    member this.BatchSetupProjects() =    
-        if openedProjects.Count > 0 then    
-            let workspace = package.ComponentModel.GetService<VisualStudioWorkspaceImpl>()
-            for siteProvider in openedProjects do
-                this.SetupProjectFile(siteProvider, workspace)
-            openedProjects.Clear()
+    override this.Initialize() =
+        base.Initialize()
+ 
+        this.Workspace.Options <- this.Workspace.Options.WithChangedOption(Completion.CompletionOptions.BlockForCompletionItems, FSharpCommonConstants.FSharpLanguageName, false)
+        this.Workspace.Options <- this.Workspace.Options.WithChangedOption(Shared.Options.ServiceFeatureOnOffOptions.ClosedFileDiagnostic, FSharpCommonConstants.FSharpLanguageName, Nullable false)
+            
+        Events.SolutionEvents.OnAfterOpenProject.Add <| fun args ->
+        async {
+            match args.Hierarchy with
+            | :? IProvideProjectSite as siteProvider ->
+                 this.SetupProjectFile(siteProvider, this.Workspace)
+            | _ -> ()
+        } |> Async.StartImmediate
```

Comment:
Right, at least until we infuse some async into `SetupProjectFile` , if it's possible at all.